File: SigreAppMobile/Hermes/app/src/main/java/pe/com/hermes/appmobile/ui/compras/ComprasOpcionesActivity.java
package pe.com.hermes.appmobile.ui.compras;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;
import pe.com.hermes.appmobile.R;
import pe.com.hermes.appmobile.data.compras.ComprasVista;
import pe.com.hermes.appmobile.data.compras.ComprasVistasCatalog;
import pe.com.hermes.appmobile.databinding.ActivityGenericListBinding;
import pe.com.hermes.appmobile.ui.common.SimpleItem;
import pe.com.hermes.appmobile.ui.common.SimpleListAdapter;

/** Hub Compras: opciones navegables (CM###). */
public class ComprasOpcionesActivity extends AppCompatActivity {

    public static final String EXTRA_GRUPO = "grupo";

    private ActivityGenericListBinding binding;
    private final List<ComprasVista> vistasVisibles = new ArrayList<>();

    public static Intent intent(Context ctx, String grupo) {
        Intent i = new Intent(ctx, ComprasOpcionesActivity.class);
        if (grupo != null) i.putExtra(EXTRA_GRUPO, grupo);
        return i;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityGenericListBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        String grupo = getIntent().getStringExtra(EXTRA_GRUPO);
        binding.toolbar.setTitle(grupo != null ? ("Compras · " + grupo) : getString(R.string.compras_opciones_titulo));
        binding.toolbar.setNavigationIcon(android.R.drawable.ic_menu_revert);
        binding.toolbar.setNavigationOnClickListener(v -> getOnBackPressedDispatcher().onBackPressed());
        binding.swipeRefresh.setEnabled(false);

        List<ComprasVista> fuente = grupo != null
                ? ComprasVistasCatalog.porGrupo(grupo)
                : ComprasVistasCatalog.navegables();
        vistasVisibles.clear();
        vistasVisibles.addAll(fuente);

        List<SimpleItem> items = new ArrayList<>();
        if (grupo == null) {
            Set<String> grupos = new LinkedHashSet<>();
            for (ComprasVista v : ComprasVistasCatalog.navegables()) grupos.add(v.grupo);
            long gid = -1;
            for (String g : grupos) {
                items.add(new SimpleItem(gid--, "▸ " + g, getString(R.string.compras_grupo_ayuda)));
            }
        }
        for (int i = 0; i < vistasVisibles.size(); i++) {
            ComprasVista v = vistasVisibles.get(i);
            items.add(new SimpleItem(i, v.nombre, v.codigoVentana + " · " + v.grupo));
        }

        binding.recyclerView.setAdapter(new SimpleListAdapter(items, item -> {
            if (item.id < 0) {
                startActivity(intent(this, item.titulo.replace("▸ ", "").trim()));
                return;
            }
            int idx = (int) item.id;
            if (idx >= 0 && idx < vistasVisibles.size()) {
                ComprasNavRouter.abrirVista(this, vistasVisibles.get(idx));
            }
        }));
    }
}

File: SigreAppMobile/SigreLogistica/app/src/main/java/pe/com/sigre/logistica/ui/compras/ComprasOpcionesActivity.kt
package pe.com.sigre.logistica.ui.compras

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import pe.com.sigre.logistica.R
import pe.com.sigre.logistica.databinding.ActivityGenericListBinding
import pe.com.sigre.logistica.ui.common.SimpleItem
import pe.com.sigre.logistica.ui.common.SimpleListAdapter
import pe.com.sigre.logistica.ui.compras.solicitudes.SolicitudesListActivity

private const val ID_SOLICITUDES = 1L

/** Submenú de Compras. Por ahora solo Solicitudes de Compra; se irán agregando Cotización/OC/OS. */
class ComprasOpcionesActivity : AppCompatActivity() {

    private lateinit var binding: ActivityGenericListBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityGenericListBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.toolbar.title = getString(R.string.compras_opciones_titulo)
        binding.toolbar.setNavigationIcon(android.R.drawable.ic_menu_revert)
        binding.toolbar.setNavigationOnClickListener { onBackPressedDispatcher.onBackPressed() }
        binding.swipeRefresh.isEnabled = false

        binding.recyclerView.layoutManager = LinearLayoutManager(this)
        binding.recyclerView.adapter = SimpleListAdapter(
            listOf(SimpleItem(ID_SOLICITUDES, getString(R.string.compras_solicitudes)))
        ) { item ->
            if (item.id == ID_SOLICITUDES) {
                startActivity(Intent(this, SolicitudesListActivity::class.java))
            }
        }
    }
}
